packages/flutter/lib/src/cupertino/page.dart

```diff
@@ -0,0 +1,173 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+const double _kMinFlingVelocity = 1.0;  // screen width per second.
+const Color _kBackgroundColor = const Color(0xFFEFEFF4); // iOS 10 background color.
+
+/// Transitions used for standard iOS full page transitions by bringing coming in from the right on
+/// top of the previous screen and by becoming pushed off-screen to the left with a parallax effect
+/// below the next screen.
+class CupertinoPageTransition extends AnimatedWidget {
+  CupertinoPageTransition({
+    Key key,
+    // Linear route animation from 0.0 to 1.0 when this screen is being pushed.
+    @required Animation<double> incomingRouteAnimation,
+    // Linear route animation from 0.0 to 1.0 when another screen is being pushed on top of this
+    // one.
+    @required Animation<double> outgoingRouteAnimation,
+    @required this.child,
+  })
+      : incomingPositionAnimation = _kRightMiddleTween.animate(
+          new CurvedAnimation(
+            parent: incomingRouteAnimation,
+            curve: Curves.easeOut,
+            reverseCurve: Curves.easeIn,
+          )
+        ),
+        outgoingPositionAnimation = _kMiddleLeftTween.animate(
+          new CurvedAnimation(
+            parent: outgoingRouteAnimation,
+            curve: Curves.easeOut,
+            reverseCurve: Curves.easeIn,
+          )
+        ),
+        super(
+          key: key,
+          // Trigger a rebuild whenever any animation route happens. The listenable's value is not
+          // used.
+          listenable: new AnimationSum(left: incomingRouteAnimation, right: outgoingRouteAnimation),
+        );
+
+  // Fractional offset from offscreen to the right to fully on screen.
+  static final FractionalOffsetTween _kRightMiddleTween = new FractionalOffsetTween(
```

Comment:
Might not fully understand dart const constructors. Don't think one is defined for fractional offset tween though.